frontend/src/metabase/query_builder/components/data-search/SearchResults.jsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+import { t } from "ttag";
+
+import Icon from "metabase/components/Icon";
+
+import Search from "metabase/entities/search";
+
+import { SearchResultItem } from "./SearchResultItem";
+
+const MAX_SEARCH_RESULTS = 200;
+
+export function SearchResults({ searchQuery, onSelect, databaseId }) {
+  const query = {
+    q: searchQuery,
+    models: ["table", "card"],
+    limit: MAX_SEARCH_RESULTS,
+    offset: 0,
+  };
+
+  if (databaseId) {
+    query["table_db_id"] = databaseId;
+  }
+
+  return (
+    <SearchResultsRoot>
+      <Search.ListLoader query={query} wrapped reload debounced>
```

Comment:
Also seems to need to handle the breaking API, which was the reason why pagination was folded into the other changes

Reply:
FE code by default supports the contract when we put data in `data` prop so there's no need to change